onnxruntime/core/platform/windows/env.cc

```diff
@@ -161,21 +156,50 @@ class WindowsThread : public EnvThread {
     }
     unsigned ret = 0;
     ORT_TRY {
-      // TODO: should I try to use SetThreadSelectedCpuSets?
       if (p->affinity.has_value() && !p->affinity->empty()) {
-        DWORD_PTR mask = 0;
-        for (auto id : *p->affinity) {
-          mask |= DWORD_PTR{1} << id;
-        }
-        auto rc = SetThreadAffinityMask(GetCurrentThread(), mask);
-        if (!rc) {
-          const auto error_code = GetLastError();
-          LOGS_DEFAULT(ERROR) << "SetThreadAffinityMask failed for thread: " << GetCurrentThreadId()
-                              << ", index: " << p->index
-                              << ", mask: " << *p->affinity
-                              << ", error code: " << error_code
-                              << ", error msg: " << std::system_category().message(error_code)
-                              << ". Specify the number of threads explicitly so the affinity is not set.";
+        int group_id = -1;
+        KAFFINITY mask = 0;
+        KAFFINITY bit = 1;
+        const WindowsEnv& env = WindowsEnv::Instance();
+        for (auto global_processor_id : *p->affinity) {
+          auto processor_info = env.GetProcessorAffinityMask(global_processor_id);
+          if (processor_info.local_processor_id > -1) {
```

Comment:
>f (processor_info.local_processor_id > -1) { [](http://example.com/codeflow?start=11&length=44)

In Posix we actually check the upper boundary. Here we assume that it is always be less than 64?

Reply:
We do not assume that - whenever "global_processor_id" is bigger than the number of logical processors in the system,  "processor_info" returned from "GetProcessorAffinityMask" will be an invalid value, meaning "processor_info.local_processor_id" will be -1.